src/client/components/request-form.js

```diff
@@ -0,0 +1,177 @@
+import { Component } from 'react';
+import h from 'react-hyperscript';
+import _ from 'lodash';
+import { makeClassList } from '../dom';
+import { truncateString } from '../../util';
+import { DOI_LINK_BASE_URL } from '../../config';
+
+const checkStatus = response => {
+  if ( response.status >= 200 && response.status < 300 ) {
+    return response;
+  } else {
+    var error = new Error( response.statusText );
+    error.response = response;
+    throw error;
+  }
+};
+
+class RequestForm extends Component {
+  constructor(props){
+    super(props);
+
+    this.bus = this.props.bus;
+
+    this.state = {
+      paperId: this.getFormField( 'paperId' ),
+      authorEmail: this.getFormField( 'authorEmail' ),
+      authorName: this.getFormField( 'authorName' ),
+      submitting: false,
+      done: false,
+      docJSON: undefined,
+      errors: {
+        incompleteForm: false,
+        network: false
+      }
+    };
+
+    this.onCloseCTA = () => this.reset();
+  }
+
+  getFormField( fieldName ){
+    const { formFields = {} } = this.props;
+    return _.get( formFields, fieldName, '' );
+  }
+
+  reset(){
+    this.setState({
+      paperId: this.getFormField( 'paperId' ),
+      authorEmail: this.getFormField( 'authorEmail' ),
+      authorName: this.getFormField( 'authorName' ),
+      submitting: false,
+      done: false,
+      docJSON: undefined,
+      errors: {
+        incompleteForm: false,
+        network: false
+      }
+    });
+  }
+
+  componentDidMount(){
+    this.bus.on('closecta', this.onCloseCTA);
+  }
+
+  componentWillUnmount(){
+    this.bus.removeListener('closecta', this.onCloseCTA);
+  }
+
+  updateForm(fields){
+    this.setState(fields);
+  }
+
+  submitRequest(){
+    const { paperId, authorEmail, authorName } = this.state;
+    const { checkIncomplete = false } = this.props;
+
+    if( !checkIncomplete && ( !paperId || !authorEmail || !authorName ) ){
```

Comment:
This would imply that author name is required in order for the form to be complete. Is this correct? 
It was always optional as it was part of the post-submit case.

Reply:
@jvwong ``checkIncomplete`` is a parameter for this component and so whether to check if any field exists is controlled by that option. Currently either all the fields can be required or optional all together controlled by ``checkIncomplete`` parameter. You mean author name should still be different then other 2 and should not be required even when ``checkIncomplete``  parameter is set?

Reply:
Oh OK that's fine.